File: schemaorg-rs/src/order_item.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///https://schema.org/additionalType
///https://schema.org/Text
///https://schema.org/URL
pub type OrderItemAdditionalTypeFieldEnum = String;
///https://schema.org/OrderItem
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct OrderItem {
    #[serde(rename = "@context")]
    pub context: String,
    ///https://schema.org/url
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///https://schema.org/alternateName
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///https://schema.org/additionalType
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<OrderItemAdditionalTypeFieldEnum>,
    ///https://schema.org/orderItemNumber
    #[serde(rename = "orderItemNumber")]
    #[serde_as(as = "OneOrMany<_>")]
    pub order_item_number: Vec<String>,
    ///https://schema.org/potentialAction
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///https://schema.org/mainEntityOfPage
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<OrderItemMainEntityOfPageFieldEnum>,
    ///https://schema.org/description
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<OrderItemDescriptionFieldEnum>,
    ///https://schema.org/identifier
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<OrderItemIdentifierFieldEnum>,
    ///https://schema.org/orderedItem
    #[serde(rename = "orderedItem")]
    #[serde_as(as = "OneOrMany<_>")]
    pub ordered_item: Vec<OrderItemOrderedItemFieldEnum>,
    ///https://schema.org/orderQuantity
    #[serde(rename = "orderQuantity")]
    #[serde_as(as = "OneOrMany<_>")]
    pub order_quantity: Vec<OrderItemOrderQuantityFieldEnum>,
    ///https://schema.org/orderDelivery
    #[serde(rename = "orderDelivery")]
    #[serde_as(as = "OneOrMany<_>")]
    pub order_delivery: Vec<ParcelDelivery>,
    ///https://schema.org/disambiguatingDescription
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///https://schema.org/image
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<OrderItemImageFieldEnum>,
    ///https://schema.org/subjectOf
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<OrderItemSubjectOfFieldEnum>,
    ///https://schema.org/orderItemStatus
    #[serde(rename = "orderItemStatus")]
    #[serde_as(as = "OneOrMany<_>")]
    pub order_item_status: Vec<OrderStatusEnum>,
    ///https://schema.org/sameAs
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///https://schema.org/name
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
}

File: schemaorg-ts/src/order_item.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type OrderItemAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/OrderItem>
 **/
export interface OrderItem {
  "@context": string;
  "@type": "OrderItem";

  /**
   * <https://schema.org/orderQuantity>
   **/
  orderQuantity: all.OneOrMany<all.OrderItemOrderQuantityFieldEnum>;

  /**
   * <https://schema.org/orderItemNumber>
   **/
  orderItemNumber: all.OneOrMany<string>;

  /**
   * <https://schema.org/orderDelivery>
   **/
  orderDelivery: all.OneOrMany<all.ParcelDelivery>;

  /**
   * <https://schema.org/orderItemStatus>
   **/
  orderItemStatus: all.OneOrMany<all.OrderStatusEnum>;

  /**
   * <https://schema.org/orderedItem>
   **/
  orderedItem: all.OneOrMany<all.OrderItemOrderedItemFieldEnum>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.OrderItemMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.OrderItemIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.OrderItemDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.OrderItemSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.OrderItemAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.OrderItemImageFieldEnum>;
}
